tests/test_simulation.py

```diff
@@ -0,0 +1,85 @@
+"""
+Copyright 2018 Grid Singularity
+This file is part of D3A.
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+import unittest
+from unittest.mock import MagicMock
+
+from d3a.d3a_core.sim_results.endpoint_buffer import SimulationEndpointBuffer
+from d3a.d3a_core.simulation import Simulation
+from d3a.models.config import SimulationConfig
+from d3a_interface.constants_limits import TIME_ZONE
+from d3a_interface.kafka_communication.kafka_producer import (DisabledKafkaConnection,
+                                                              KafkaConnection)
+from d3a_interface.sim_results.all_results import ResultsHandler
+from pendulum import duration, today
+
+
+class SimulationTest(unittest.TestCase):
+
+    @staticmethod
+    def test_results_not_send_via_kafka_if_started_from_cli():
+        redis_job_id = None
+        simulation_config = SimulationConfig(duration(hours=int(12)),
+                                             duration(minutes=int(60)),
+                                             duration(seconds=int(60)),
+                                             market_count=1,
+                                             cloud_coverage=0,
+                                             market_maker_rate=30,
+                                             start_date=today(tz=TIME_ZONE),
+                                             external_connection_enabled=False)
+        simulation = Simulation(
+            "default_2a", simulation_config, None, 0, False, duration(), False, False, None, None,
+            redis_job_id, False
+        )
+        simulation.endpoint_buffer = MagicMock(spec=SimulationEndpointBuffer)
+        results_mapping = ResultsHandler().results_mapping
+        simulation.endpoint_buffer.results_handler = MagicMock(spec=ResultsHandler)
+        simulation.kafka_connection = MagicMock(spec=DisabledKafkaConnection)
+        simulation.endpoint_buffer.results_handler.all_ui_results = \
+            {k: {} for k in results_mapping}
```

Comment:
If you want to follow PEP-8, you can avoid using the backslash (which is discouraged) and do this:
```suggestion
        simulation.endpoint_buffer.results_handler.all_ui_results = {
            k: {} for k in results_mapping}
```
I don't know why PyCharm doesn't enforce this yet. They have an open ticket for it anyway (https://youtrack.jetbrains.com/issue/PY-17193).